lib/NetPacksLib.cpp

```diff
@@ -411,7 +407,7 @@ DLL_LINKAGE void RemoveObject::applyGs(CGameState *gs)
 
 	if(obj->ID == Obj::HERO) //remove beaten hero
 	{
-		CGHeroInstance * beatenHero = static_cast<CGHeroInstance*>(obj);
+		auto * beatenHero = dynamic_cast<CGHeroInstance *>(obj);

```

Comment:
Why should we replace static_cast with dynamic one? This is questionable. We are not prepared to nullptr anyway?

Reply:
I do agree with change.
Yes, now code *may* result in `beatenHero == nullptr`, but old code was no safe anyway.
We might drop assert or even abort if this happens, but this is still better than old code.

Using static_cast for upcasting is NOT safe, and will only result in crash at some place later. So better to detect this immediately.
```suggestion
		auto * beatenHero = dynamic_cast<CGHeroInstance *>(obj);
		assert(beatenHero != nullptr);
```

Reply:
This is more about intention. If you use static_cast you are sure what type is while if you use dynamic - you are not sure. As I understand c++ has no way to give some meaningful error in case of invalid static_cast so option with dynamic cast and assert may look better

Reply:
done assert